app/models/patient.rb

```diff
@@ -129,6 +129,136 @@ def latest_transfer
+  }
+
+  # Non-eligible contact methods are: ['Unknown', 'Opt-out', '', nil]
+  scope :has_usable_preferred_contact_method, lambda {
+    where.not(preferred_contact_method: ['Unknown', 'Opt-out', '', nil])
+  }
+
+  # Check if the HoH has any dependents that make them eligible to
+  # recieve notifications on the dependent's behalf.
+  # The joined table is referred to as `dependents_patients` and is used for all
+  # checks made on the dependents. Anywhere the `patients` table is specified or
+  # where a hash is used for a condition are checks made on the HoH.
+  #
+  # A dependent patient makes the HoH reminder eligible if:
+  # - HoH is not purged
+  # - HoH has a usable preferred contact method
+  # - HoH has not paused notifications
+  # - current HoH local time is within the preferred contact time window
+  # - patient has monitoring ON
+  # - patient is not purged
+  # - patient is in isolation
+  #   OR
+  #   patient is in continuous exposure
+  #   OR
+  #   patient last date of exposure is on or after (today - monitoring_period_days)
+  #   OR
+  #   patient last date of exposure is null and created at of exposure is on or after (today - monitoring_period_days)
+  scope :hoh_eligible_for_any_dependents, lambda {
+    joins(:dependents)
+      .where(purged: false)
+      .where(head_of_household: true)
+      .where('patients.id = patients.responder_id')
+      .where(
+        # Ignore any joined rows where the dependents_patients is the HoH itself.
+        'dependents_patients.id != dependents_patients.responder_id'
+      )
+      .has_usable_preferred_contact_method
+      .where(
+        # HoH is unconditionally ineligible if it has paused notifications
+        pause_notifications: false
+      )
+      .where('dependents_patients.monitoring = ?', true)
+      .where('dependents_patients.purged = ?', false)
+      .where(
+        'dependents_patients.isolation = ? '\
+        'OR dependents_patients.continuous_exposure = ? '\
+        'OR dependents_patients.last_date_of_exposure >= DATE_SUB(DATE(CONVERT_TZ(?, "UTC", dependents_patients.time_zone)), INTERVAL ? DAY) '\
+        'OR ('\
+        '  dependents_patients.last_date_of_exposure IS NULL '\
+        '  AND dependents_patients.created_at >= DATE_SUB(DATE(CONVERT_TZ(?, "UTC", dependents_patients.time_zone)), INTERVAL ? DAY)'\
+        ')',
+        true,
+        true,
+        Time.now.getlocal('-00:00'),
+        ADMIN_OPTIONS['monitoring_period_days'],
+        Time.now.getlocal('-00:00'),
+        ADMIN_OPTIONS['monitoring_period_days']
+      )
+      .within_preferred_contact_time
+  }
+
+  # Patients should be monitored are any of the below:
+  # - In isolation
+  # - In continuous exposure
+  # - last date of exposure is on or after (today - monitoring_period_days)
+  #   OR
+  #   last date of exposure is null and created at of exposure is on or after (today - monitoring_period_days)
+  scope :should_be_monitoring, lambda {
```

Comment:
I originally decided against `within_monitoring_period` because it seemed like it implied that it would not include patients that are not in time-boxed monitoring plans (like isolation and continuous exposure).

Reply:
Yeah I had that thought as well. Thoughts on `is_being_monitored` or better yet, `is_monitored`?

Reply:
changed to `is_being_monitored`